src/main/scala/is/hail/variant/VariantDataset.scala

```diff
@@ -504,7 +504,7 @@ class VariantDatasetFunctions(private val vds: VariantSampleMatrix[Genotype]) ex
   }
 
   def hardCalls(): VariantDataset = {
-    vds.mapValues { g => Genotype(g.gt, g.fakeRef) }
+    vds.mapValues { g => Genotype(g.gt, g.fakeRef) }.copy(isLinearScale = false)
```

Comment:
Not sure I like this.  Why not `Genotype(g.gt, g.fakeRef, vds.isLinearScale)`?

`VDS.isLinearScale` should be used for read/writing genotypes only, since we don't serialize `GenotypeStream.isLinearScale` when writing a VDS.

Where does `hardcalls().write()` fail for BGENs?